mono/utils/mono-merp.c

```diff
@@ -359,7 +359,7 @@ mono_init_merp (const intptr_t crashed_pid, const char *signal, MonoStackHash *h
 	merp->systemManufacturer = "apple";
 	get_apple_model ((char *) merp->systemModel, sizeof (merp->systemModel));
 
-	merp->eventType = config.eventType;
+	merp->eventType = "MonoAppCrash";
```

Comment:
This comes from the `eventType` argument to the call to `EnableMicrosoftTelemetry` in managed, which is called by VSfM. I would prefer that we update the event type there instead of hardcoding something in here.

Reply:
The interesting bit of this is that the layout of the XML file here (with its specific parameters in a specific order) is deeply tied to the `EventType` given to Watson.  It actually doesn't make sense to have VSfM be able to control that value, since if you mis-match the two you will get weird behavior from Watson...which we don't want.

I think it would make more sense to remove the `EventType` parameter from the pinvoke that VSfM calls into Mono with to remove that parameter and actually hard code it here as a result...  Though changing that pinvoke signature means we have to be sure to update the two in parallel?

In any case, I leave it to you all to work out the right information here, but I did want to add some context around why it makes sense for mono to control the `eventType` here itself and not let VSfM change it: Because the bucketing parameters/ordering/values are deeply tied to the `eventType`...you can't drop any other name into `eventType` other than `MonoAppCrash` and expect anything other than terrible results.

Reply:
Glad you brought it up...

So that causes a bit of a problem - if the parameters provided do not match what the back-end expects for a given `eventType` it will get rejected.  I am not sure why the the previous implementation put the XML payload in Mono, and the event type in VSfM when they _need_ to match.  AFAIU, there is no a scenario where the `eventType` can be set to something different and the upload still work.

If I update the parameters here, then we run the risk of an old VSM (with the old event type) running on this newer Mono and all it's events rejected.  Same goes the other way, if we change it in VSM and it's running on Mono from before this change, all the events will be rejected.  And so on...

Of course, we can coordinate the changes to land simultaneously, but that doesn't stop creative users from mix-matching things. 

There is no good answer to be honest, because with my change, the `EnableMicrosoftTelemetry` now has a parameter for `eventType` that get's ignored. And changing that signature is probably worse than just ignoring the parameter. 

I decided that the best solution to minimize rejected events, and prevent this scenario in the future, was to set the event type in the same place (meaning Mono/MERP) that we construct the XML/parameters (and ignore the one set by VSM). But of course, I am open to other suggestions and discussion how to best go forward (even if you think we should leave it as is, it just doesn't make sense to me 😛)

Reply:
GH should have collaborative editing...

Reply:
I see.  That makes sense. Thanks @leculver and @kdubau


@kdubau I'm going to push some updates to this PR as well:
1. Add a comment summarizing why we're hardcoding the event type.
2. Update `EnableMicrosoftTelemetry` to call the parameter `unused` instead of `eventType`
3. I'll update the internal call signatures (`EnableMicrosoftTelemetry_internal`) to drop the `eventType` parameter entirely.

Reply:
Sounds good, I'm still testing these changes and may have more depending how that goes.